### DISCUSSION TITLE
Kendala Lubridate dalam R Studio

### DISCUSSION THREAD

**Discussion by DeanAlexander27** [teamalgoritma/community#535] [Q&A] [answered]
Dear Team Algoritma

Saya menemukan kendala ketika in class DV Ultron B untuk Day Class
dimana dapat dilihat bahwa ada kendala di line 251 yang seharusnya menyesuaikan jam dengan waktu NY dengan perbedaan waktu kurang lebih 5 jam kebelakang dari waktu JKT, meskipun terkadang bisa dengan run all chunk berkali kali. 

saya sudah coba menggunakan kodingan force_tz tetapi tidak sesuai ouput yang diarahkan oleh mentoring. apa ada solusi ?
![image](https://user-images.githubusercontent.com/123348110/217132987-c3665c32-e176-4764-bd1f-9a507c991144.png)

**Comment by jafarhf1** [comment, marked as answer]
Dear Pak Dean,

Terkait kendala yang bapak alami, hal itu disebabkan karena pak Dean sudah melakukan assignment ketika di awal pertama kali merubah tipe datanya tanpa memasukkan parameter tz (baris ke-238), sehingga data `vids$publish_time` nya sudah berubah tipe datanya dengan timezone UTC. Untuk merubah timezone dari tipe data yang sudah berupa datetime/POSIXct agar jamnya sesuai dengan yang kita inginkan maka bisa menggunakan fungsi `with_tz()`. contohnya:
`with_tz(vids$publish_time, tz="America/New_York")`

Demikian yang dapat kami sampaikan, Semoga membantu.

**Reply by DeanAlexander27** [reply]
Thank you ka Jafar.. iya ini baru sadar pas coba run chunk berulang ternyata bermasalah nya di bagian baris 238 assign di awal. saya udah mencoba pakai with_tz() pun ketika di run berulang outputnya berulang,, sehingga saya memutuskan untuk tidak assign di awal di baris 238, dan sekarang outpunya sudah sesuai, terima kasih ka.
